### PR TITLE
Project/admit IDs query for mobile app

### PR DESCRIPTION
**Pull request by spamhurts**
New query to select a list of admit/project IDs for a given animal on a given event date.

### CODE REVIEW

File: snprc_ehr/resources/queries/study/availableChargeIds.sql [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+   	ANIMAL_ID VARCHAR,
+    EVENT_DATE TIMESTAMP
+)
+
+-- Research Charge IDs
+SELECT
+    CAST(p.project AS VARCHAR(40)) AS admitProjectId,
+    p.project AS projectId,
+    sp.projectType AS projectType,
+    0 AS admitId,
```
Comment (RameshRapa) [inline snprc_ehr/resources/queries/study/availableChargeIds.sql:11]: 0 As admitId. are we setting 0 here Terry?

File: snprc_ehr/resources/queries/study/availableChargeIds.sql [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+WHERE aaa.id = ANIMAL_ID
+  AND GREATEST (p.startDate, aaa.date, sp.startDate) <= EVENT_DATE
+  AND LEAST (p.endDate, TIMESTAMPADD('SQL_TSI_DAY', 1, aaa.endDate), TIMESTAMPADD('SQL_TSI_DAY', 1, sp.endDate), NOW()) >= EVENT_DATE
+-- Maintenance/Behavior Charge IDs
+UNION
+SELECT DISTINCT
+    CAST(p.project AS varchar(40)) AS admitProjectId,
+    p.project AS projectId,
+    sp.projectType AS projectType,
+    0 AS admitId,
```
Comment (RameshRapa) [inline snprc_ehr/resources/queries/study/availableChargeIds.sql:29]: 0 as aditId.

File: snprc_ehr/resources/queries/study/availableChargeIds.sql [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+-- Research Charge IDs
+SELECT
+    CAST(p.project AS VARCHAR(40)) AS admitProjectId,
+    p.project AS projectId,
+    sp.projectType AS projectType,
+    0 AS admitId,
+    sp.description AS projectText,
+    GREATEST (p.startDate, aaa.date, sp.startDate) AS startDate,
+    LEAST (p.endDate, TIMESTAMPADD('SQL_TSI_DAY', 1, aaa.endDate), TIMESTAMPADD('SQL_TSI_DAY', 1, sp.endDate), NOW()) AS endDate,
```
Comment (RameshRapa) [inline snprc_ehr/resources/queries/study/availableChargeIds.sql:14]: Can i please know what is this "SQL_TSI_DAY"? just for my understanding.

Comment (spamhurts) [reply snprc_ehr/resources/queries/study/availableChargeIds.sql:14]: It tells the timestampadd function to use days as the unit to add to a date. Check out the LK SQL Reference: https://www.labkey.org/Documentation/wiki-page.view?name=labkeysql

**Review by RameshRapa — APPROVED** [review]

**Review by jallentxbiomed — APPROVED** [review]

**Pull request merged**